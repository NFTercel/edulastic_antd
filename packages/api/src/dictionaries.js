import API from "./utils/API";

const api = new API();

const receiveCurriculums = () => api.callApi({ url: "/curriculum" }).then(result => result.data.result);

const receiveStandards = ({ curriculumId, grades = [], search }) => {
  const data = { curriculumId, grades, search };
  return api
    .callApi({
      method: "post",
      url: "/search/standards",
      data
    })
    .then(result =>
      result.data.result.hits.hits.map(el => ({ _id: el._id, ...el._source })).filter(item => item.level === "ELO")
    );
};

export default {
  receiveCurriculums,
  receiveStandards
};
