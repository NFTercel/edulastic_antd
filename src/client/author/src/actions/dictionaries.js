import {
  RECEIVE_DICT_CURRICULUMS_REQUEST,
  RECEIVE_DICT_STANDARDS_REQUEST,
  CLEAR_DICT_STANDARDS
} from "../constants/actions";

export const getDictCurriculumsAction = () => ({
  type: RECEIVE_DICT_CURRICULUMS_REQUEST
});

export const getDictStandardsForCurriculumAction = (curriculumId, grades, search) => ({
  type: RECEIVE_DICT_STANDARDS_REQUEST,
  payload: { curriculumId, grades, search }
});

export const clearDictStandardsAction = () => ({
  type: CLEAR_DICT_STANDARDS
});
