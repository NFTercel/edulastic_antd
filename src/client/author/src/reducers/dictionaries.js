import {
  RECEIVE_DICT_CURRICULUMS_REQUEST,
  RECEIVE_DICT_CURRICULUMS_SUCCESS,
  RECEIVE_DICT_CURRICULUMS_ERROR,
  RECEIVE_DICT_STANDARDS_REQUEST,
  RECEIVE_DICT_STANDARDS_SUCCESS,
  RECEIVE_DICT_STANDARDS_ERROR,
  CLEAR_DICT_STANDARDS
} from "../constants/actions";

const initialItemsState = {
  curriculums: {
    curriculums: [],
    loading: false,
    error: null
  },
  standards: {
    standards: [],
    loading: false,
    error: null
  }
};

const dictionariesReducer = (state = initialItemsState, { type, payload }) => {
  switch (type) {
    case RECEIVE_DICT_CURRICULUMS_REQUEST:
      return {
        ...state,
        curriculums: {
          ...state.curriculums,
          loading: true
        }
      };
    case RECEIVE_DICT_CURRICULUMS_SUCCESS:
      return {
        ...state,
        curriculums: {
          ...state.curriculums,
          curriculums: payload.items,
          loading: false
        }
      };
    case RECEIVE_DICT_CURRICULUMS_ERROR:
      return {
        ...state,
        curriculums: {
          ...state.curriculums,
          loading: false,
          error: payload.error
        }
      };
    case RECEIVE_DICT_STANDARDS_REQUEST:
      return {
        ...state,
        standards: {
          ...state.standards,
          loading: true
        }
      };
    case RECEIVE_DICT_STANDARDS_SUCCESS:
      return {
        ...state,
        standards: {
          ...state.standards,
          standards: payload.items,
          loading: false
        }
      };
    case RECEIVE_DICT_STANDARDS_ERROR:
      return {
        ...state,
        standards: {
          ...state.standards,
          loading: false,
          error: payload.error
        }
      };
    case CLEAR_DICT_STANDARDS:
      return {
        ...state,
        standards: {
          ...state.standards,
          standards: []
        }
      };
    default:
      return state;
  }
};

export default dictionariesReducer;
