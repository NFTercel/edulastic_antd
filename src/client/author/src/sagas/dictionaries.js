import { takeLatest, call, put, all } from "redux-saga/effects";
import { dictionariesApi } from "@edulastic/api";
import { message } from "antd";
import {
  RECEIVE_DICT_CURRICULUMS_REQUEST,
  RECEIVE_DICT_CURRICULUMS_SUCCESS,
  RECEIVE_DICT_CURRICULUMS_ERROR,
  RECEIVE_DICT_STANDARDS_REQUEST,
  RECEIVE_DICT_STANDARDS_SUCCESS,
  RECEIVE_DICT_STANDARDS_ERROR
} from "../constants/actions";

function* receiveCurriculumsSaga() {
  try {
    const items = yield call(dictionariesApi.receiveCurriculums);

    yield put({
      type: RECEIVE_DICT_CURRICULUMS_SUCCESS,
      payload: { items }
    });
  } catch (err) {
    console.error(err);
    const errorMessage = "Receive curriculums is failing";
    yield call(message.error, errorMessage);
    yield put({
      type: RECEIVE_DICT_CURRICULUMS_ERROR,
      payload: { error: errorMessage }
    });
  }
}

function* receiveStandardsSaga({ payload }) {
  try {
    const items = yield call(dictionariesApi.receiveStandards, payload);

    yield put({
      type: RECEIVE_DICT_STANDARDS_SUCCESS,
      payload: { items }
    });
  } catch (err) {
    console.error(err);
    const errorMessage = "Receive standards is failing";
    yield call(message.error, errorMessage);
    yield put({
      type: RECEIVE_DICT_STANDARDS_ERROR,
      payload: { error: errorMessage }
    });
  }
}

export default function* watcherSaga() {
  yield all([
    yield takeLatest(RECEIVE_DICT_CURRICULUMS_REQUEST, receiveCurriculumsSaga),
    yield takeLatest(RECEIVE_DICT_STANDARDS_REQUEST, receiveStandardsSaga)
  ]);
}
