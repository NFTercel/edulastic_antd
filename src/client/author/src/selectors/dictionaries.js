import { createSelector } from "reselect";

export const stateSelector = state => state.dictionaries;
export const curriculumsSelector = createSelector(
  stateSelector,
  state => state.curriculums
);
export const getCurriculumsListSelector = createSelector(
  curriculumsSelector,
  state => state.curriculums
);

export const standardsSelector = createSelector(
  stateSelector,
  state => state.standards
);
export const getStandardsListSelector = createSelector(
  standardsSelector,
  state => state.standards
);
