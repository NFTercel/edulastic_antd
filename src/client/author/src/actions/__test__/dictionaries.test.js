import { getDictCurriculumsAction, getDictStandardsForCurriculumAction } from "../dictionaries";

describe("Dictionaries", () => {
  it("get DicCurriculums should return an action", () => {
    expect(getDictCurriculumsAction()).toMatchSnapshot();
  });

  it("get DicStandardsForCurriculum should return an action", () => {
    expect(getDictStandardsForCurriculumAction("5e26tw6q21", 5, "search")).toMatchSnapshot();
  });
});
