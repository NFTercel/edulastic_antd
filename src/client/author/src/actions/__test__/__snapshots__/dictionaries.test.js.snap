// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dictionaries get DicCurriculums should return an action 1`] = `
Object {
  "type": "[dictionaries] receive curriculums request",
}
`;

exports[`Dictionaries get DicStandardsForCurriculum should return an action 1`] = `
Object {
  "payload": Object {
    "curriculumId": "5e26tw6q21",
    "grades": 5,
    "search": "search",
  },
  "type": "[dictionaries] receive standards request",
}
`;
